src/badge/Badge.tsx

```diff
@@ -18,21 +18,67 @@ export type BadgeProps = BoxProps & {
    * @default "primary"
    */
   variant?: keyof Renderlesskit.GetThemeValue<"badge", "variant">;
+  /**
+   * floats the badge on parent element's corners
+   *
+   * @default false
+   */
+  floating?: boolean;
+  /**
+   * floating position
+   *
+   * @default "top-right"
+   */
+  position?: keyof Renderlesskit.GetThemeValue<"badge", "position">;
 };
 
 export const Badge = forwardRefWithAs<BadgeProps, HTMLSpanElement, "span">(
   (props, ref) => {
-    const { variant = "primary", size = "md", className, ...rest } = props;
+    const htmlref = React.useRef<HTMLSpanElement>();
+    const {
+      position = "top-right",
+      variant = "primary",
+      size = "md",
+      floating = false,
+      className,
+      ...rest
+    } = props;
 
     const theme = useTheme();
     const badgeStyles = cx(
       theme.badge.base,
       theme.badge.size[size],
       theme.badge.variant[variant],
+      floating
+        ? cx(
+            theme.badge.attached,
+            theme.badge.position[position],
+            !props.children ? theme.badge.dot[size] : "",
+          )
+        : "",
       className,
     );
 
-    return <Box as="span" ref={ref} className={badgeStyles} {...rest} />;
+    React.useEffect(() => {
+      if (!floating) return;
+      if (ref && htmlref.current) {
```

Comment:
I don't think its necessary to check both the ref 😑

Reply:
thats a mistake